Apps/Src/Task_FaultState.c

```diff
@@ -31,26 +31,27 @@ void EnterFaultState() {
     // Turn LEDs On and logs Error into EEPROM
     BSP_Light_Off(RUN); //turn off run light
     BSP_Light_On(FAULT);
-    SafetyStatus voltStatus = Voltage_CheckStatus();
-    if (voltStatus == OVERVOLTAGE){
-        BSP_Light_On(OVOLT);
-        EEPROM_LogError(FAULT_HIGH_VOLT);
-    }
-    if (voltStatus == UNDERVOLTAGE){
-        BSP_Light_On(UVOLT);
-        EEPROM_LogError(FAULT_LOW_VOLT);
-    }
-    if (Temperature_CheckStatus(Amps_IsCharging()) == DANGER){
-        BSP_Light_On(OTEMP);
-        EEPROM_LogError(FAULT_HIGH_TEMP);
-    }
-    if (Amps_CheckStatus(false) == DANGER){
-        BSP_Light_On(OCURR);
-        EEPROM_LogError(FAULT_HIGH_CURRENT);
-    }
-    if (BSP_WDTimer_DidSystemReset()){
-        BSP_Light_On(WDOG);
-        EEPROM_LogError(FAULT_WATCHDOG);
+    switch (Fault_BitMap){
+        case 1:
+            BSP_Light_On(UVOLT);
+            EEPROM_LogError(FAULT_LOW_VOLT);
```

Comment:
`EEPROM_LogError()` yields because it calls the I2C BSP. I am working on fixing the I2C BSP on a separate branch though, so I can fix this. My idea is to set it up so we can force the system to use the bare metal version of the BSP if we are in a fault state. What do you think @manthanand @chaseblock @sugamarora99

Reply:
I think it's a good idea.